velox/connectors/hive/storage_adapters/s3fs/tests/S3FileSystemTest.cpp

```diff
@@ -344,3 +345,36 @@ TEST_F(S3FileSystemTest, logLevel) {
   config["hive.s3.log-level"] = "FATAl";
   checkLogLevelName("FATAL");
 }
+
+TEST_F(S3FileSystemTest, writeFileAndRead) {
+  const auto bucketName = "writedata";
+  const auto file = "test.txt";
+  const auto filename = localPath(bucketName) + "/" + file;
+  const auto s3File = s3URI(bucketName, file);
+
+  auto hiveConfig = minioServer_->hiveConfig();
+  filesystems::S3FileSystem s3fs(hiveConfig);
+  s3fs.initializeClient();
+  auto writeFile = s3fs.openFileForWrite(s3File);
+  std::string dataContent =
```

Comment:
Yes! there should be a lot more tests for various paths. I updated the description.
We can probably make `kMinPartSize`, `kMaxPartSize` configurable since different object stores have different limits.
Alibaba cloud says part must be `[100KB to 5GB]` https://www.alibabacloud.com/help/en/oss/user-guide/multipart-upload-12

Reply:
I added a test for the multi-part upload.

Reply:
you can add a TODO for partsize configuration.